src/smc-webapp/frame-editors/latex-editor/actions.ts

```diff
@@ -449,6 +483,43 @@ export class Actions extends BaseActions<LatexEditorState> {
     }
   }
 
+  async run_pythontex(time: number, force: boolean): Promise<void> {
+    let output: BuildLog;
+    const status = s => this.set_status(`Running PythonTeX... ${s}`);
+    status("");
+
+    try {
+      // Run PythonTeX
+      output = await pythontex(this.project_id, this.path, time, force, status);
```

Comment:
there is no need to compute a sha1 hash like with sagetex (to avoid running sage), because pythontex runs much quicker and  decides if it does any work based on new modifications and has also tracks file dependencies on its own, etc.